File: VB/DevExpress.Example04/Data/Parent.vb
Imports Microsoft.VisualBasic
Imports DevExpress.Mvvm
Imports System.Collections.ObjectModel

Namespace DevExpress.Example04.Data

	Public Class Toy
		Inherits BindableBase
		Protected _Name As String
		Public Property Name() As String
			Get
				Return Me._Name
			End Get
			Set(ByVal value As String)
				Me.SetProperty(Me._Name, value, "Name")
			End Set
		End Property

	End Class

	Public Class Child
		Inherits BindableBase

		Protected _Name As String

		Public Property Name() As String
			Get
				Return Me._Name
			End Get
			Set(ByVal value As String)
				Me.SetProperty(Me._Name, value, "Name")
			End Set
		End Property

		Protected _Age As Integer
		Public Property Age() As Integer
			Get
				Return Me._Age
			End Get
			Set(ByVal value As Integer)
				Me.SetProperty(Me._Age, value, "Age")
			End Set
		End Property

		Protected _IsExpanded As Boolean

		Public Property IsExpanded() As Boolean
			Get
				Return Me._IsExpanded
			End Get
			Set(ByVal value As Boolean)
				Me.SetProperty(Me._IsExpanded, value, "IsExpanded")
			End Set
		End Property

		Protected _Toys As ObservableCollection(Of Toy)

		Public ReadOnly Property Toys() As ObservableCollection(Of Toy)
			Get
				If Me._Toys Is Nothing Then
					Me._Toys = New ObservableCollection(Of Toy)()
				End If

				Return Me._Toys
			End Get
		End Property
	End Class

	Public Class Parent
		Inherits Child

		Protected _Children As ObservableCollection(Of Child)

		Public ReadOnly Property Children() As ObservableCollection(Of Child)
			Get
				If Me._Children Is Nothing Then
					Me._Children = New ObservableCollection(Of Child)()
				End If
				Return Me._Children
			End Get
		End Property
	End Class
End Namespace

File: CS/DevExpress.Example04/Data/Parent.cs
using DevExpress.Mvvm;
using System.Collections.ObjectModel;

namespace DevExpress.Example04.Data {

    public class Toy : BindableBase {
        protected string _Name;
        public string Name {
            get { return this._Name; }
            set { this.SetProperty(ref this._Name, value, "Name"); }
        }

    }

    public class Child : BindableBase {
        
        protected string _Name;

        public string Name {
            get { return this._Name; }
            set { this.SetProperty(ref this._Name, value, "Name"); }
        }

        protected int _Age;
        public int Age {
            get { return this._Age; }
            set { this.SetProperty(ref this._Age, value, "Age"); }
        }

        protected bool _IsExpanded;

        public bool IsExpanded {
            get { return this._IsExpanded; }
            set { this.SetProperty(ref this._IsExpanded, value, "IsExpanded"); }
        }

        protected ObservableCollection<Toy> _Toys;

        public ObservableCollection<Toy> Toys {
            get {
                if(this._Toys == null) {
                    this._Toys = new ObservableCollection<Toy>();
                }

                return this._Toys;
            }
        }
    }
    
    public class Parent : Child {

        protected ObservableCollection<Child> _Children;

        public ObservableCollection<Child> Children {
            get {
                if(this._Children == null) {
                    this._Children = new ObservableCollection<Child>();
                }
                return this._Children;
            }
        }
    }
}
